### PR TITLE
docs(readme): add usage section

### PR DESCRIPTION
**Pull request by smmoosavi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +53,105 @@ Add [proxy-polyfill](https://github.com/GoogleChrome/proxy-polyfill) to support
...
+```
+
+And create a wire with the `useWire` hook and pass it to components:
+
+```tsx
+function FancyList(props) {
+  const filter$ = useWire(null, '');
+  return (
+    <>
+      <FnacyInput value$={filter$} />
```
Comment (MahdiAbdi) [inline README.md:126]:
```suggestion
      <FancyInput value$={filter$} />
```

File: README.md [hunk 2]
```diff
@@ -52,6 +53,105 @@ Add [proxy-polyfill](https://github.com/GoogleChrome/proxy-polyfill) to support
...
+  return (
+    <>
+      <FnacyInput value={filter} setValue={setFilter} />
+      <List items={props.items} filter={filter} />
+    </>
+  );
+}
+```
+
+- you need to refactor all `FancyInput` usage, pass `value`, and `setValue` to make it work as expected
```
Comment (MahdiAbdi) [inline README.md:95]:
```suggestion
- you need to refactor all the `FancyInput` usages, pass `value`, and `setValue` to make it work as expected.
```

File: README.md [hunk 3]
```diff
@@ -52,6 +53,105 @@ Add [proxy-polyfill](https://github.com/GoogleChrome/proxy-polyfill) to support
...
+- the top-level component (`FancyList`) re-render each time the `filter` changes
```
Comment (MahdiAbdi) [inline README.md:96]:
```suggestion
- the top-level component (`FancyList`) re-renders each time the `filter` changes.
```

File: README.md [hunk 4]
```diff
@@ -52,6 +53,105 @@ Add [proxy-polyfill](https://github.com/GoogleChrome/proxy-polyfill) to support
...
+```tsx
+function FancyList(props) {
+  const filter$ = useWire(props.filter$, '');
+  /* ... */
+}
+```
+
+react-wire have more advanced features:
+
+- selectors to maintain calculated values
```
[the remark this answers is not in the document]
Comment (smmoosavi) [reply README.md:150]: For now, Yes. I will add an example directory in the future.

**Pull request merged**